crates/modelardb_manager/src/main.rs

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2023 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Implementation of the ModelarDB manager main function.
+
+use object_store::ObjectStore;
+use std::env;
+use std::sync::Arc;
+
+use modelardb_common::arguments::{
+    argument_to_remote_object_store, collect_command_line_arguments,
+};
+use modelardb_common::remote_data_folder::validate_remote_data_folder_from_argument;
+use sqlx::postgres::PgConnectOptions;
+use sqlx::{ConnectOptions, PgConnection};
+use tokio::runtime::Runtime;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+use crate::remote::start_apache_arrow_flight_server;
+
+mod remote;
+
+pub const PORT: u16 = 8888;
+
+/// Parse the command line arguments to extract the metadata database and the remote object store
+/// and start an Apache Arrow Flight server. Returns [`String`] if the command line arguments
+/// cannot be parsed, if the metadata cannot be read from the database, or if the Apache Arrow
+/// Flight server cannot be started.
+fn main() -> Result<(), String> {
+    // Initialize a tracing layer that logs events to stdout.
+    let stdout_log = tracing_subscriber::fmt::layer();
+    tracing_subscriber::registry().with(stdout_log).init();
+
+    // Create a Tokio runtime for executing asynchronous tasks.
+    let runtime = Arc::new(
+        Runtime::new().map_err(|error| format!("Unable to create a Tokio Runtime: {error}"))?,
+    );
+
+    let arguments = collect_command_line_arguments(3);
```

Comment:
It seems that returning `Vec<&str>` is technically possible but it would involve adding explicit lifetimes so the general recommendation is to just return `Vec<String>`. Since this seems to be the standard, I have not added a comment.

Reply:
Ok.